Examples/Python/src/GeometryBuildingGen1.cpp

```diff
@@ -0,0 +1,155 @@
+            .def(py::init<Acts::LayerCreator::Config>())
+            .def("cylinderLayer",
+                 [](const Acts::LayerCreator &self, const GeometryContext &gctx,
+                    SurfacePtrVector surfaces, std::size_t binsPhi,
+                    std::size_t binsZ) {
+                   return self.cylinderLayer(gctx, surfaces, binsPhi, binsZ);
+                 })
+            .def("discLayer",
+                 [](const Acts::LayerCreator &self, const GeometryContext &gctx,
+                    SurfacePtrVector surfaces, std::size_t binsR,
+                    std::size_t binsPhi) {
+                   return self.discLayer(gctx, surfaces, binsR, binsPhi);
+                 });
+
+    auto config =
+        py::class_<LayerCreator::Config>(creator, "Config").def(py::init<>());
+    ;
+
+    ACTS_PYTHON_STRUCT_BEGIN(config, LayerCreator::Config);
+    ACTS_PYTHON_MEMBER(surfaceArrayCreator);
+    ACTS_PYTHON_MEMBER(cylinderZtolerance);
+    ACTS_PYTHON_MEMBER(cylinderPhiTolerance);
+    ACTS_PYTHON_STRUCT_END();
+  }
+
+  {
+    using Creator = Acts::SurfaceArrayCreator;
+    using Config = typename Creator::Config;
+
+    auto creator =
+        py::class_<Creator, std::shared_ptr<Creator>>(m, "SurfaceArrayCreator")
+            .def(py::init<Config>());
+
+    py::class_<Config>(creator, "Config").def(py::init<>());
+  }
+
+  {
+    using Base = Acts::ILayerArrayCreator;
+    using Creator = Acts::LayerArrayCreator;
+    using Config = typename Creator::Config;
+
+    py::class_<Base, std::shared_ptr<Base>>(m, "ILayerArrayCreator");
+
+    auto creator = py::class_<Creator, std::shared_ptr<Creator>, Base>(
+                       m, "LayerArrayCreator")
+                       .def(py::init<Config>());
+
+    py::class_<Config>(creator, "Config").def(py::init<>());
+  }
+
+  {
+    using Base = Acts::ITrackingVolumeArrayCreator;
+    using Creator = Acts::TrackingVolumeArrayCreator;
+    using Config = typename Creator::Config;
+
+    py::class_<Base, std::shared_ptr<Base>>(m, "ITrackingVolumeArrayCreator");
+
+    auto creator = py::class_<Creator, std::shared_ptr<Creator>, Base>(
+                       m, "TrackingVolumeArrayCreator")
+                       .def(py::init<Config>());
+
+    py::class_<Config>(creator, "Config").def(py::init<>());
+  }
+
+  py::class_<Acts::VolumeBounds, std::shared_ptr<Acts::VolumeBounds>>(
+      m, "VolumeBounds");
+
+  py::class_<Acts::CylinderVolumeBounds,
+             std::shared_ptr<Acts::CylinderVolumeBounds>, Acts::VolumeBounds>(
+      m, "CylinderVolumeBounds")
+      .def(py::init<ActsScalar, ActsScalar, ActsScalar, ActsScalar, ActsScalar,
+                    ActsScalar, ActsScalar>(),
+           "rmin"_a, "rmax"_a, "halfz"_a, "halfphi"_a = M_PI, "avgphi"_a = 0.,
+           "bevelMinZ"_a = 0., "bevelMaxZ"_a = 0.);
+
+  {
+    // NOTE Here we break the interface down from a transform to a zshift, since
+    // this is the only thing used
```

Comment:
Is that necessary? Generally, I find it to be the least surprising if the Python version does not introduce extra logic over C++, whenever that's possible (which it isn't always of course)